art/defences/trainer/adversarial_trainer_trades_pytorch.py

```diff
@@ -126,8 +126,13 @@ def fit(
             # compute accuracy
             if validation_data is not None:
                 (x_test, y_test) = validation_data
+
                 output = np.argmax(self.predict(x_test), axis=1)
-                nb_correct_pred = np.sum(output == np.argmax(y_test, axis=1))
+                if y_test.ndim > 1:
```

Comment:
The idea of this change is good. Let's follow the ART pattern by using `y = check_and_transform_label_format(y, nb_classes=self._classifier.nb_classes)` for `y` and `validation_data[1]` at the beginning of this method to ensure they are both one-hot-encoded.